File: Source/Game/Management/ProgressPointHelper.cs
using System.Collections.Generic;
using WCSharp.Api;
using static WCSharp.Api.Common;

public class ProgressPointHelper
{
    public static List<RectPoints> Points;
    public int CurrentPoint;

    public ProgressPointHelper()
    {
        Points ??= RectPoints.InitPoints();
        CurrentPoint = 0;
    }
}

public class RectPoints
{
    public float X { get; set; }
    public float Y { get; set; }

    public RectPoints(float x, float y)
    {
        X = x;
        Y = y;
    }

    public static List<RectPoints> InitPoints()
    {
        List<RectPoints> points = new List<RectPoints>();
        foreach (var point in RegionList.PathingPoints)
        {
            points.Add(new RectPoints(point.Rect.CenterX, point.Rect.CenterY));
        }
        return points;
    }
}

File: src/Game/Management/ProgressPointHelper.ts
import { RegionList } from 'src/Global/RegionList'

export class ProgressPointHelper {
    public static Points: RectPoints[]
    public CurrentPoint = 0

    public constructor() {
        ProgressPointHelper.Points ??= RectPoints.InitPoints()
        this.CurrentPoint = 0
    }
}

export class RectPoints {
    public x = 0
    public y = 0

    public constructor(x: number, y: number) {
        this.x = x
        this.y = y
    }

    public static InitPoints(): RectPoints[] {
        let points: RectPoints[] = []
        for (let point of RegionList.PathingPoints) {
            points.push(new RectPoints(point.centerX, point.centerY))
        }
        return points
    }
}
